SU2_CFD/src/fluid/CNonPolytropicPengRobinson.cpp

```diff
@@ -0,0 +1,582 @@
+/*!
+ * \file CNonPolytropicPengRobinson.cpp
+ * \brief Source of the non-polytropic Peng-Robinson model.
+ * \author B. Fuentes Monjas
+ * \version 7.3.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2022, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../include/fluid/CNonPolytropicPengRobinson.hpp"
+
+CNonPolytropicPengRobinson::CNonPolytropicPengRobinson(su2double gamma, su2double R, su2double Pstar, su2double Tstar, su2double w, const CConfig *config)
+    : CPengRobinson (gamma, R, Pstar, Tstar, w) {
+  Temperature = TstarCrit; // Initialization of Temperature to critical value
+}
+
+void CNonPolytropicPengRobinson::SetCpModel(const CConfig* config) {
+  for (int i = 0; i < config->GetnPolyCoeffs(); ++i) {
+    coeffs_[i] = config->GetCp_PolyCoeffND(i);
+  }
+}
+
+su2double CNonPolytropicPengRobinson::T_v_h(su2double v, su2double h){
+  su2double toll = 1e-9, FACTOR = 0.2;
+  su2double f, fv, C0, C1, C2, C4, C6, C8, C10, T, d, atanh;
+  su2double x1, x2, xmid, dx, fx1, fx2, fmid = 1.0, rtb;
+  su2double sqrt2 = sqrt(2.0);
+  unsigned short countrtb = 0, NTRY = 100, ITMAX = 100;
+
+  d = (v * v + 2 * b * v - b * b);
+
+  atanh = (log(1.0 + (b * sqrt2 / (v + b))) - log(1.0 - (b * sqrt2 / (v + b)))) / 2.0;
```

Comment:
don't use the name of a function as a variable, also, why didn't you use the `atanh` function directly? it's in the <cmath> header